example-app/src/main/java/com/adyen/checkout/example/repositories/paymentMethods/PaymentsRepository.kt

```diff
@@ -47,15 +50,34 @@ internal class PaymentsRepositoryImpl(private val checkoutApiService: CheckoutAp
     }
 
     override fun paymentsRequest(paymentsRequest: PaymentsRequest): Call<ResponseBody> {
-        return checkoutApiService.payments(paymentsRequest)
+        return checkoutApiService.payments(paymentsRequest.combineToJSONObject())
     }
 
     override suspend fun paymentsRequestAsync(paymentsRequest: PaymentsRequest): ResponseBody? {
         return safeApiCall(
-            call = { checkoutApiService.paymentsAsync(paymentsRequest) }
+            call = { checkoutApiService.paymentsAsync(paymentsRequest.combineToJSONObject()) }
         )
     }
 
+    private fun PaymentsRequest.combineToJSONObject(): JSONObject {
+        val moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
+        val adapter = moshi.adapter(PaymentsRequestData::class.java)
+        val requestDataJson = JSONObject(adapter.toJson(this.requestData))
+
+        return this.paymentComponentData
```

Comment:
I had this comment in `PaymentRequestAdapter`:
`paymentComponentData should be merged second to override any possible duplicated fields`
So I think we should `return this.requestDataJson.putAll(paymentComponentData)`?

Reply:
Good catch!